service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/ManageDocumentsUploadedEventHandler.java

```diff
@@ -0,0 +1,341 @@
+
+        Map<Set<String>, Function<DocumentUploadedNotificationConfiguration, ConfidentialLevel>> resultMap =
+            new HashMap<>();
+
+        // designated LA
+        Set<String> designatedLA = furtherEvidenceNotificationService
+            .getDesignatedLocalAuthorityRecipientsOnly(caseData);
+        if (designatedLA.isEmpty()) {
+            log.info("No recipient found for designated LA");
+        } else {
+            resultMap.put(designatedLA,
+                DocumentUploadedNotificationConfiguration::getSendToDesignatedLA);
+        }
+
+        // secondary LA
+        Set<String> secondaryLA = furtherEvidenceNotificationService.getSecondaryLocalAuthorityRecipientsOnly(caseData);
+        if (secondaryLA.isEmpty()) {
+            log.info("No recipient found for secondary LA");
+        } else {
+            resultMap.put(secondaryLA,
+                DocumentUploadedNotificationConfiguration::getSendToSecondaryLA);
+        }
+
+        // legal representative
+        Set<String> legalRepresentative = furtherEvidenceNotificationService.getLegalRepresentativeOnly(caseData);
+        if (legalRepresentative.isEmpty()) {
+            log.info("No recipient found for legal representative");
+        } else {
+            resultMap.put(legalRepresentative,
+                DocumentUploadedNotificationConfiguration::getSendToLegalRepresentative);
+        }
+
+        // cafcass representative
+        Set<String> cafcassRepresentative = furtherEvidenceNotificationService.getCafcassRepresentativeEmails(caseData);
+        if (cafcassRepresentative.isEmpty()) {
+            log.info("No recipient found for cafcass representative");
+        } else {
+            resultMap.put(cafcassRepresentative,
+                DocumentUploadedNotificationConfiguration::getSendToCafcassRepresentative);
+        }
+
+        // respondent solicitor
+        Set<String> respondentSolicitor = furtherEvidenceNotificationService.getRespondentSolicitorEmails(caseData);
+        if (respondentSolicitor.isEmpty()) {
+            log.info("No recipient found for respondent solicitor");
+        } else {
+            resultMap.put(respondentSolicitor,
+                DocumentUploadedNotificationConfiguration::getSendToRespondentSolicitor);
+        }
+
+        // child solicitor
+        Set<String> childSolicitor = furtherEvidenceNotificationService.getChildSolicitorEmails(caseData);
+        if (childSolicitor.isEmpty()) {
+            log.info("No recipient found for child solicitor");
+        } else {
+            resultMap.put(childSolicitor,
+                DocumentUploadedNotificationConfiguration::getSendToChildSolicitor);
+        }
+
+        return resultMap;
+    }
+
+    private boolean isHearingDocument(DocumentType documentType) {
+        return COURT_BUNDLE.equals(documentType) || CASE_SUMMARY.equals(documentType)
+               || POSITION_STATEMENTS.equals(documentType) || SKELETON_ARGUMENTS.equals(documentType);
```

Comment:
Shall we add a static method to DocumentType?

```
    public static Set<DocumentType> hearingDocuments() {
        return Set.of(COURT_BUNDLE, CASE_SUMMARY, POSITION_STATEMENTS, SKELETON_ARGUMENTS);
    }
```

And change this method to `return DocumentType.hearingDocuments.contains(documentType)`